packages/core/src/__tests__/index.test.tsx

```diff
@@ -1486,3 +1486,75 @@ it("doesn't throw if children is null", () => {
 
   expect(() => render(element).update(element)).not.toThrowError();
 });
+
+function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+it('set proper title', async () => {
+  const TestNavigator = (props: any) => {
+    const { state, descriptors } = useNavigationBuilder(MockRouter, props);
+
+    return descriptors[state.routes[state.index].key].render();
+  };
+
+  let navigate: () => void = () => {};
+
+  const TestScreen = (to: string) => (props: any) => {
+    navigate = () =>
```

Comment:
I think the test might be more clear if the `to` parameter is passed to `navigate`